sound/soc/sof/pm.c

```diff
@@ -234,6 +234,10 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 	struct snd_sof_dev *sdev = dev_get_drvdata(&priv->pdev_pcm->dev);
 	int ret = 0;
 
+	/* do nothing if the probe failed somehow */
+	if (!sdev)
+		return 0;
```

Comment:
@plbossart Thanks, yes this covers all our discussions on this topic but I have a question here. If sdev is in fact NULL, should we return 0 or an error? 0 indicates that suspend/resume was successful. But what we're doing here is avoiding a suspend/resume in this case.